src/Core/Domain/CatalogPriceRule/QueryResult/CatalogPriceRuleForListing.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\QueryResult;
+
+use DateTimeInterface;
+use PrestaShop\Decimal\DecimalNumber;
+
+class CatalogPriceRuleForListing
+{
+    /**
+     * @var int
+     */
+    private $catalogPriceRuleId;
+
+    /**
+     * @var string
+     */
+    private $catalogPriceRuleName;
+
+    /**
+     * @var string|null
+     */
+    private $shopName;
+
+    /**
+     * @var string|null
+     */
+    private $currencyName;
+
+    /**
+     * @var string|null
+     */
+    private $countryName;
+
+    /**
+     * @var string|null
+     */
+    private $groupName;
+
+    /**
+     * @var int
+     */
+    private $fromQuantity;
+
+    /**
+     * @var string
+     */
+    private $reductionType;
+
+    /**
+     * @var DecimalNumber
+     */
+    private $reduction;
+
+    /**
+     * @var DateTimeInterface
+     */
+    private $dateStart;
+
+    /**
+     * @var DateTimeInterface
+     */
+    private $dateEnd;
+
+    public function __construct(
+        int $catalogPriceRuleId,
+        string $catalogPriceRuleName,
+        ?string $shopName,
+        ?string $currencyName,
+        ?string $countryName,
+        ?string $groupName,
```

Comment:
Please put the nullable parameters a the end of the parameters list, it feels more natural to have the mandatory ones first Even if in the end all parameters must be provided, at least you can identify the optional ones in one look (ideally same for class fields en getters)